api-contracts/v1/workflows.proto

```diff
@@ -0,0 +1,163 @@
+enum StickyStrategy {
+    SOFT = 0;
+    HARD = 1;
+}
+
+enum RateLimitDuration {
+    SECOND = 0;
+    MINUTE = 1;
+    HOUR = 2;
+    DAY = 3;
+    WEEK = 4;
+    MONTH = 5;
+    YEAR = 6;
+}
+
+// CreateWorkflowVersionRequest represents options to create a workflow version.
+message CreateWorkflowVersionRequest {
+    string name = 1; // (required) the workflow name
+    string description = 2; // (optional) the workflow description
+    string version = 3; // (optional) the workflow version
+    repeated string event_triggers = 4; // (optional) event triggers for the workflow
+    repeated string cron_triggers = 5; // (optional) cron triggers for the workflow
+    repeated CreateTaskOpts tasks = 6; // (required) the workflow jobs
+    Concurrency concurrency = 7; // (optional) the workflow concurrency options
+    optional string schedule_timeout = 8; // (optional) the timeout for the schedule
+    optional string cron_input = 9; // (optional) the input for the cron trigger
+    optional CreateTaskOpts on_failure_task = 10; // (optional) the job to run on failure
+    optional StickyStrategy sticky = 11; // (optional) the sticky strategy for assigning steps to workers
+    optional int32 default_priority = 12; // (optional) the priority of the workflow
+}
+
+enum ConcurrencyLimitStrategy {
+    CANCEL_IN_PROGRESS = 0;
+    DROP_NEWEST = 1; // deprecated
+    QUEUE_NEWEST = 2; // deprecated
+    GROUP_ROUND_ROBIN = 3;
+    CANCEL_NEWEST = 4;
+}
+
+message Concurrency {
+    string expression = 1; // (required) the expression to use for concurrency
+    optional int32 max_runs = 2; // (optional) the maximum number of concurrent workflow runs, default 1
+    optional ConcurrencyLimitStrategy limit_strategy = 3; // (optional) the strategy to use when the concurrency limit is reached, default CANCEL_IN_PROGRESS
+}
+
+enum WorkerLabelComparator {
+    EQUAL = 0;
+    NOT_EQUAL = 1;
+    GREATER_THAN = 2;
+    GREATER_THAN_OR_EQUAL = 3;
+    LESS_THAN = 4;
+    LESS_THAN_OR_EQUAL = 5;
+}
+
+message DesiredWorkerLabels {
+    // value of the affinity
+    optional string strValue = 1;
+    optional int32 intValue = 2;
+
+    /**
+    * (optional) Specifies whether the affinity setting is required.
+    * If required, the worker will not accept actions that do not have a truthy affinity setting.
+    *
+    * Defaults to false.
+    */
+    optional bool required = 3;
+
+    /**
+    * (optional) Specifies the comparator for the affinity setting.
+    * If not set, the default is EQUAL.
+    */
+    optional WorkerLabelComparator comparator = 4;
+
+    /**
+    * (optional) Specifies the weight of the affinity setting.
+    * If not set, the default is 100.
+    */
+    optional int32 weight = 5;
+}
+
+// CreateTaskOpts represents options to create a task.
+message CreateTaskOpts {
+    string readable_id = 1; // (required) the task name
+    string action = 2; // (required) the task action id
+    string timeout = 3; // (optional) the task timeout
```

Comment:
is schedule time out not on the task now?

Reply:
nice catch, it is

Reply:
maybe schedule_timeout on the workflow should be changed to default_schedule timeout

also priority is task level